src/Symfony/Component/DependencyInjection/EnvVarProcessor.php

```diff
@@ -185,6 +187,20 @@ public function getEnv($prefix, $name, \Closure $getEnv)
             return $env;
         }
 
+        if ('yaml' === $prefix) {
+            try {
+                $env = Yaml::parse($env);
```

Comment:
What kind of runtime checks do I have to add here? 

The YAML component isn't a requirement for the DependencyInjection one.